File: LeetCode/Medium/2448-count-number-of-bad-pairs/count-number-of-bad-pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        n: int = len(nums)
        answer: int = n * (n - 1) // 2
        frequency: dict[int, int] = {}

        for idx, val in enumerate(nums):
            key = idx - val
            answer -= frequency.get(key, 0)
            frequency[key] = frequency.get(key, 0) + 1

        return answer
        

File: LeetCode/Medium/2448-count-number-of-bad-pairs/count-number-of-bad-pairs.rs
use std::collections::HashMap;

impl Solution {
    pub fn count_bad_pairs(nums: Vec<i32>) -> i64 {
        let mut answer = (nums.len() * (nums.len() - 1) / 2) as i64;
        let mut frequency = HashMap::<i32, usize>::new();

        for (index, value) in nums.iter().enumerate() {
            let key = index as i32 - *value;
            *frequency.entry(key).or_insert(0) += 1;
        }

        for value in frequency.values() {
            if *value > 1 {
                answer -= (*value * (*value - 1) / 2) as i64;
            }
        }

        answer
    }
}